var/spack/repos/builtin/packages/sz/package.py

```diff
@@ -133,4 +135,16 @@ def cmake_args(self):
             args.append("-DBUILD_STATS=ON")
         else:
             args.append("-DBUILD_STATS=OFF")
+
+        if self.run_tests:
+            args.append("-DBUILD_TESTS=ON")
+            args.append("-DINTEGRATION_TESTS=ON")
+        else:
+            args.append("-DBUILD_TESTS=OFF")
+            args.append("-DINTEGRATION_TESTS=OFF")
         return args
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
```

Comment:
Adding these makes this a build-time test, not a "stand-alone/smoke test" that can be run weeks to months after the package is installed.

Since that's the context, the name of the method needs to be changed (e.g., `test_install`).

For this to be a stand-alone/smoke test, it will need to be available outside of the build.  There's more on that topic at https://spack.readthedocs.io/en/latest/packaging_guide.html#stand-alone-or-smoke-tests.

Reply:
It looks reasonable. I agree on this change. Thanks.

Reply:
@disheng222 this will require some more work to ensure the testing files are available per the docs she mentioned.